File: src/gleamgen_ffi.erl
-module(gleamgen_ffi).
-export([identity/1, get_function_name/1, get_matcher_output/1]).

identity(X) -> X.

get_function_name(Funct) ->
  {name, Name} = erlang:fun_info(Funct, name),
  erlang:atom_to_binary(Name).

get_matcher_output({_, _, nil}) -> {error, nil};
get_matcher_output({_, _, X}) -> {ok, X}.

File: src/gleamgen_ffi.mjs
import { Ok, Error } from "./gleam.mjs";

export function identity(x) {
  return x;
}

export function get_function_name(x) {
  return x.name;
}

export function get_matcher_output(x) {
  if (x.output) {
    return new Ok(x.output);
  } else {
    return new Error(undefined);
  }
}
